### PR TITLE
REGR: Series[dt64/td64].astype(string)

### LINKED ISSUE #41409: REGR: Change to Series[dt64/td64].astype("string")` behavior on master
xref https://github.com/pandas-dev/pandas/pull/38562#issuecomment-836438843 cc @jbrockmendel 

old behaviour

```
>>> pd.__version__
'1.2.4'
>>> 
>>> tdi = pd.timedelta_range("1 Day", periods=3)
>>> ser = pd.Series(tdi)
>>> ser
0   1 days
1   2 days
2   3 days
dtype: timedelta64[ns]
>>> ser.astype("string")
0    1 days
1    2 days
2    3 days
dtype: string
>>> 
>>> dti = pd.date_range("2021", periods=3)
>>> ser = pd.Series(dti)
>>> ser
0   2021-01-01
1   2021-01-02
2   2021-01-03
dtype: datetime64[ns]
>>> ser.astype("string")
0    2021-01-01
1    2021-01-02
2    2021-01-03
dtype: string
```

new behaviour

```
>>> pd.__version__
'1.3.0.dev0+1567.g67c9385787'
>>> 
>>> tdi = pd.timedelta_range("1 Day", periods=3)
>>> ser = pd.Series(tdi)
>>> ser
0   1 days
1   2 days
2   3 days
dtype: timedelta64[ns]
>>> ser.astype("string")
0     86400000000000 nanoseconds
1    172800000000000 nanoseconds
2    259200000000000 nanoseconds
dtype: string
>>> 
>>> dti = pd.date_range("2021", periods=3)
>>> ser = pd.Series(dti)
>>> ser
0   2021-01-01
1   2021-01-02
2   2021-01-03
dtype: datetime64[ns]
>>> ser.astype("string")
0    2021-01-01T00:00:00.000000000
1    2021-01-02T00:00:00.000000000
2    2021-01-03T00:00:00.000000000
dtype: string
>>> 
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #41409
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

This surfaces another bug in `StringArray.__getitem__` that I haven't looked into yet.

### CODE REVIEW

File: pandas/_libs/lib.pyx [hunk 1]
```diff
@@ -712,6 +712,14 @@ cpdef ndarray[object] ensure_string_array(
         Py_ssize_t i = 0, n = len(arr)
 
     if hasattr(arr, "to_numpy"):
+
+        if hasattr(arr, "dtype") and arr.dtype.kind in ["m", "M"]:
```
Comment (jreback) [inline pandas/_libs/lib.pyx:716]: agreed, yeah the `.to_numpy()` on an EA is the approparate here.

File: pandas/tests/frame/methods/test_astype.py [hunk 2]
```diff
@@ -660,6 +656,15 @@ def test_astype_dt64_to_string(self, frame_or_series, tz_naive_fixture, request)
         alt = obj.astype(str)
         assert np.all(alt.iloc[1:] == result.iloc[1:])
 
+    def test_astype_td64_to_string(self, frame_or_series):
```
Comment (jreback) [inline pandas/tests/frame/methods/test_astype.py:659]: can you parametersize over td, dt, dt w/tz & period

Comment (jbrockmendel) [reply pandas/tests/frame/methods/test_astype.py:659]: not nicely; the dt64 one uses tz_naive_fixture

Comment (jreback) [reply pandas/tests/frame/methods/test_astype.py:659]: ok can you add that as another test then. just to make sure coverng the bases.

File: pandas/_libs/lib.pyx [hunk 3]
```diff
@@ -716,6 +716,14 @@ cpdef ndarray[object] ensure_string_array(
         Py_ssize_t i = 0, n = len(arr)
 
     if hasattr(arr, "to_numpy"):
```
Comment (jreback) [inline pandas/_libs/lib.pyx:718]: we dont' have access to the ExtnsionTypes here right? so this is just as easy

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/methods/test_astype.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/frame/methods/test_astype.py:659]: add what as another case?  we have the td64 case here and the dt64 case right above here

File: pandas/_libs/lib.pyx (hunk 3 above)
Comment (jbrockmendel) [reply pandas/_libs/lib.pyx:718]: right

File: pandas/tests/frame/methods/test_astype.py (hunk 2 above)
Comment (jreback) [reply pandas/tests/frame/methods/test_astype.py:659]: ahh ok. 

period covered? if not pls add (followup ok)

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
@meeseeksdev backport 1.3.x

**Pull request merged**